### ISSUE TITLE
Performance los

### ISSUE THREAD

**Issue by vadimlitvinenko87** [Eclipse-Community/r3dfox-old#22]
Tested on Windows 7 x64, RAM: 16GB DDR3, CPU: Intel i5-3570 4Ghz, GPU: Radeon RX 570 4GB,
https://browserbench.org/

r3dfox-122.0.1.en-US.win64 (v122.0.1-2)
J - 102.000
M - 793
S - 153

r3dfox-123.0.2.en-US.win64 (Aero v123.0.2-2)
J - 95.500
M - 444
S - 152

**Comment by vadimlitvinenko87** [comment]
Motion Mark tested much lower perf. JetStream bit lower. Not good tendention.

**Comment by the-r3dacted** [comment]
Can you see if this has the same performance regression?

https://github.com/Alex313031/Mercury/releases/

I followed their performance changes for my browser and enabled cross thinLTO. I'll also publish a build with normal LTO and see what happens.

**Comment by the-r3dacted** [comment]
Made a new build

https://github.com/Eclipse-Community/r3dfox/releases/tag/123.0.2-2-beta-aero

**Comment by vadimlitvinenko87** [comment]
[r3dfox-123.0.2.en-US.win64.installer.exe](https://github.com/Eclipse-Community/r3dfox/releases/download/123.0.2-2-beta-aero/r3dfox-123.0.2.en-US.win64.installer.exe)
 58 MB
 created 27 minutes ago
old J - 95.500 new J - 104.850 much better then v122.0.1
old M - 444    new M - 449 looks equal to old
old S - 152     new  S - 154 looks equal to old
[JetStream 2.pdf](https://github.com/Eclipse-Community/r3dfox/files/14563953/JetStream.2.pdf)

**Comment by the-r3dacted** [comment]
Closing since this should be done

**Comment by vadimlitvinenko87** [comment]
Motion mark performance los not patched, but jetstream good score, looks like need solve problem with motion mark too to close issue.

**Comment by vadimlitvinenko87** [comment]
I am not pushing just want to help.

**Comment by the-r3dacted** [comment]
Oh you said looks equal to old which I assumed meant 122 and didn't check. Not sure what would've caused this. Can you check Mercury on 10 to see if it's any different.

Also check about:support on both and tell me what is compositing pls.

**Comment by vadimlitvinenko87** [comment]
Mercury release on Win10?
Or redfox 123?

**Comment by the-r3dacted** [comment]
> Mercury release on Win10? Or redfox 123?

Yeah both Mercury 122 and 123 on 10. Also try r3dfox 122 and 123 on 10 too since that may show something too.

**Comment by vadimlitvinenko87** [comment]
Ok, will check

**Comment by vadimlitvinenko87** [comment]
[r3dfox 122 W10.pdf](https://github.com/Eclipse-Community/r3dfox/files/14564702/r3dfox.122.W10.pdf)
[r3dfox 123 W10.pdf](https://github.com/Eclipse-Community/r3dfox/files/14564703/r3dfox.123.W10.pdf)

[Redfox v122 W10 - JetStream 101.689.pdf](https://github.com/Eclipse-Community/r3dfox/files/14564704/Redfox.v122.W10.-.JetStream.101.689.pdf)
[Redfox v122 W10 - MotionMark 806 points.pdf](https://github.com/Eclipse-Community/r3dfox/files/14564705/Redfox.v122.W10.-.MotionMark.806.points.pdf)
[Redfox v123 W10 - JetStream 105.541 points.pdf](https://github.com/Eclipse-Community/r3dfox/files/14564706/Redfox.v123.W10.-.JetStream.105.541.points.pdf)
[Redfox v123 W10 - MotionMark 754 points.pdf](https://github.com/Eclipse-Community/r3dfox/files/14564707/Redfox.v123.W10.-.MotionMark.754.points.pdf)

[Merc 123 AVX W10.pdf](https://github.com/Eclipse-Community/r3dfox/files/14564708/Merc.123.AVX.W10.pdf)
[Merc 123 SSE4 W10.pdf](https://github.com/Eclipse-Community/r3dfox/files/14564709/Merc.123.SSE4.W10.pdf)

[Merc v123 AVX W10 - JetStream 104.246 points.pdf](https://github.com/Eclipse-Community/r3dfox/files/14564710/Merc.v123.AVX.W10.-.JetStream.104.246.points.pdf)
[Merc v123 AVX W10 - MotionMark 711 points.pdf](https://github.com/Eclipse-Community/r3dfox/files/14564711/Merc.v123.AVX.W10.-.MotionMark.711.points.pdf)
[Merc v123 SSE4 W10 - JetStream 106.428 points.pdf](https://github.com/Eclipse-Community/r3dfox/files/14564712/Merc.v123.SSE4.W10.-.JetStream.106.428.points.pdf)
[Merc v123 SSE4 W10 - MotionMark 710 points.pdf](https://github.com/Eclipse-Community/r3dfox/files/14564701/Merc.v123.SSE4.W10.-.MotionMark.710.points.pdf)

[r3dfox 122 W7.pdf](https://github.com/Eclipse-Community/r3dfox/files/14564759/r3dfox.122.W7.pdf)
[r3dfox 123 W7.pdf](https://github.com/Eclipse-Community/r3dfox/files/14564757/r3dfox.123.W7.pdf)

[Redfox v122 W7 - MotionMark 786 points.pdf](https://github.com/Eclipse-Community/r3dfox/files/14564896/Redfox.v122.W7.-.MotionMark.786.points.pdf)
[Redfox v123 W7 - MotionMark 445 points.pdf](https://github.com/Eclipse-Community/r3dfox/files/14564897/Redfox.v123.W7.-.MotionMark.445.points.pdf)

**Comment by vadimlitvinenko87** [comment]
Windows 7 tests will be tomorrow.

**Comment by vadimlitvinenko87** [comment]
[Redfox v123v3 W7 - JetStream 105.500.pdf](https://github.com/Eclipse-Community/r3dfox/files/14567463/Redfox.v123v3.W7.-.JetStream.105.500.pdf)
[Redfox v123v3 W7 - MotionMark 449.pdf](https://github.com/Eclipse-Community/r3dfox/files/14567464/Redfox.v123v3.W7.-.MotionMark.449.pdf)
[r3dfox 123v3 W7.pdf.pdf](https://github.com/Eclipse-Community/r3dfox/files/14567465/r3dfox.123v3.W7.pdf.pdf)
MotionMark on last version not better. Once i had v122 lose performance before reboot, but after some boots and reinstalling 122-123 nothing changed as was, anyway v122 faster +78% in MotionMark.

**Comment by vadimlitvinenko87** [comment]
Windows7 x64 browsers in BrowserBench.org tests аfter issue closed.
![SCRN-20240313151531](https://github.com/Eclipse-Community/r3dfox/assets/34354361/fcea7050-8802-4fff-9cbd-c4183f16340c)

**Comment by the-r3dacted** [comment]
Pushed a new build, let me know if it's improved.

https://github.com/Eclipse-Community/r3dfox/releases/tag/123.0.2-3-beta

**Comment by vadimlitvinenko87** [comment]
[Beta v123.0.2-3 x64 WebRender fix W7 - MotionMark 749 points.pdf](https://github.com/Eclipse-Community/r3dfox/files/14584477/Beta.v123.0.2-3.x64.WebRender.fix.W7.-.MotionMark.749.points.pdf)
[Beta v123.0.2-3 x64 WebRender fix W7 -JetStream 105.000.pdf](https://github.com/Eclipse-Community/r3dfox/files/14584479/Beta.v123.0.2-3.x64.WebRender.fix.W7.-JetStream.105.000.pdf)
Performance in motion mark much better, looks like problem solved. Thanks for work.
P.s. Speedometer3.0 (third test) updated on BrowserBench.org and Beta.v123.0.2-3.x64.WebRender.fix shows 9.5 points now (for example CentBrowser v118 made same 9 points)

**Comment by the-r3dacted** [comment]
Maybe somewhat related, but I was testing on HD 4600 and 123 seems to drop MotionMark score a substantial amount with hardware acceleration.

**Comment by J4CKED** [comment]
When I found out that there is a new version of Firefox for windows 7, I was happy. 
R3dfox is fast to open and responsive, but I noticed that the scrolling of long pages is not as smooth as in Firefox 113.0.2 (64-bit) and it is noticeable.

I also did a test of another version
https://github.com/adeii/supermium-portable/releases/download/F1235/Firefox-portable-126.0b2-win32.7z

and it is the same. Well, and now I'm saddened, because it follows that it can't be fixed.  That it is simply the fault of changes to Firefox itself.

**Comment by the-r3dacted** [comment]
Maybe could be tweaked with about:config. Can I see a video of what it looks like?

**Comment by J4CKED** [comment]
I recorded the video, but as I see youtube will probably not display it in 60 fps. Firefox from 0 to 13 seconds, and then r3dfox.
https://www.youtube.com/watch?v=mnMcQD5SlIM

Anyway, when scrolling through pages where there are a lot of elements (gif images) the page does not scroll as smoothly as in the old firefox. This is especially annoying when reading text and scrolling slowly.

**Comment by the-r3dacted** [comment]
I can't really see it. Can you check stock 115 and see how it is? Also try older r3dfox versions too.

**Comment by J4CKED** [comment]
Version 115.10.0esr (64-bit) is very smooth. I downloaded it from PortableApps.com because I prefer portable versions.
Old versions of r3dfox is not smooth.

**Comment by the-r3dacted** [comment]
So even earliest 121 has the same problem?

**Comment by J4CKED** [comment]
I tested this version. I don't see an earlier one. 
https://github.com/Eclipse-Community/r3dfox/releases/download/122.0.1-beta/firefox-122.0.1.en-US.win64.installer.exe

**Comment by the-r3dacted** [comment]
Just made a new release with a bunch of test builds, see if it still happens as far down as 117.

**Comment by J4CKED** [comment]
Appreciate strongly.
I first downloaded and installed firefox version-117.0.2

At first when I scrolled the test page it was visible lag. But after a while it disappeared and the page started scrolling very smoothly. I was very surprised.

After a while I opened the latest version (already configured for myself) r3dfox-125.0.2 and here also the page scrolled smoothly.

In addition, I don't know what happened, but in this version 125.0.2 the minimize, maximize and close buttons disappeared, although they were still normally visible in the morning. To fix this, I had to change the windows style to a different one. However, in Firefox 115.10.0 the buttons are visible in any windows style.

At this point r3dfox is running smoothly. If there are any changes I will let you know.
